tasks/updatelibrary.js

```diff
@@ -1,17 +1,64 @@
 'use strict';
 
+const fs = require('fs');
 const path = require('path');
+const crypto = require('crypto');
+const existsSync = require('exists-sync');
+function hash (contents) {
+	return crypto.createHash('md5').update(contents).digest('hex');
+}
+
+function constructPolyfill(polyfillPaths) {
+	const polyfills = polyfillPaths.map((polyfill) => fs.readFileSync(polyfill));
+	return polyfills.join('');
+}
+
+const installPolyfill = function(polyfillOutputFolder, module, polyfillSourcePaths) {
+	const polyfillOutputPath = path.join(polyfillOutputFolder, 'polyfill.js');
+	const polyfillAlreadyExists = existsSync(polyfillOutputPath);
+
+	polyfillSourcePaths = polyfillSourcePaths || [''];
+	polyfillSourcePaths = polyfillSourcePaths.map(polyfillPath => path.join(module, polyfillPath))
+	polyfillSourcePaths = polyfillSourcePaths.map(path => require.resolve(path));
+
+	console.log(`Importing polyfill/s from ${polyfillSourcePaths.join(',')}`);
+	const newPolyfill = constructPolyfill(polyfillSourcePaths);
+
+	if (polyfillAlreadyExists) {
+		const currentPolyfill = fs.readFileSync(polyfillOutputPath);
+		const currentPolyfillHash = hash(currentPolyfill);
+		const newPolyfillHash = hash(newPolyfill);
+
+		if (newPolyfillHash === currentPolyfillHash) {
+			console.log('Polyfill has not changed.');
+			return;
+		} else {
+			console.log('Polyfill has changed. Deleting old polyfill.');
+			fs.unlinkSync(polyfillOutputPath);
+		}
+  }
+
+	fs.writeFileSync(polyfillOutputPath, newPolyfill);
+	console.log('Polyfill imported successfully');
+
+};
 
 module.exports = function(grunt) {
 
 	grunt.registerMultiTask('updatelibrary', 'Update Polyfill imported from external libraries', function() {
 		this.files.forEach(function (file) {
-			file.src.filter(function(src) {
-				return src.indexOf('/__versions/') === -1;
-			}).forEach(function(src){
-				grunt.log.writeln('Running ' + src + ': ');
-				require(path.resolve(src.toString()))(grunt);
+			const configs = file.src.map(src => Object.assign({src}, {
+				config: JSON.parse(fs.readFileSync(src)),
+			}));
+			configs.filter(function(config) {
+				return typeof config.config.module === 'string';
```

Comment:
So here I think it's more appropriate to be looking for a section, like `update`